File: Python/Array/Minimum no. of Jumps to reach end of an array.py
	def minJumps(self, arr, n):
	    #code here
	    if n==1:
	        return 0
	    jump,end,maxind=0,0,0
	    for i in range(0,n-1):
	        maxind=max(maxind,arr[i]+i)
	        if i==end:
	            jump+=1
	            end=maxind
	        if end>=n-1:
	            return jump
	    return -1

File: Cpp/Array/Minimum no. of Jumps to reach end of an array.cpp
    int minJumps(int arr[], int n){
        // Your code here
        int maxind=0,jump=0,end=0;
        if(n==1)
            return 0;
        for(int i=0;i<n-1;i++){
            maxind=max(arr[i]+i,maxind);
            if(end==i){
                jump++; 
                end=maxind;
            }
            if(end>=n-1)
                return jump;
        }
        return -1;
    }